src/main/java/edu/texas/social_computing/hospitals/HRP.java

```diff
@@ -1,9 +1,81 @@
 package edu.texas.social_computing.hospitals;
 
-import java.util.Queue;
+import java.util.*;
 
 public class HRP {
+    private HRP() {}
+
+    /**
+     * From: The Hospitals / Residents Problem (1962; Gale, Shapley)
+     * David F. Manlove
+     * https://pdfs.semanticscholar.org/77d9/f84082674888ca90ca662847983381b23338.pdf?_ga=2.216999441.622816506.1541285838-70410261.1541285838
+     *
+     * M := ∅;
+     * while (some resident r_i is unassigned and r_i has a non-empty list) {
+     *     h_j := first hospital on r_i’s list;
+     *     // r_i applies to h_j
+     *     M := M ∪ {(r_i, h_j)};
+     *     if (h_j is over-subscribed) {
+     *             r_k := worst resident in M(h_j) according to h_j’s list;
+     *             M := M\{(r_k, h_j)};
+     *     }
+     *     if (h_j is full) {
+     *             r_k := worst resident in M(h_j) according to h_j’s list;
+     *             for (each successor r_l of r_k on h_j’s list)
+     *             delete the pair (r_l, h_j);
+     *     }
+     * }
+     */
+
     public static Matching run(HospitalTable hospitalTable, ResidentTable residentTable, Queue<Resident> freeResidents) {
-        return null;
+        Iterator<Resident> freeResidentsItir = freeResidents.iterator();
```

Comment:
I don't think there is a need for this iterator ... just use `freeResidents.poll()`

Reply:
I agree!